backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/outputGenerators/DocxGenerator.java

```diff
@@ -48,28 +50,33 @@ public DocxGenerator(OutputFormatVariant outputFormatVariant, String description
         ByteArrayOutputStream docxOutputStream = new ByteArrayOutputStream();
-        try {
-            XWPFDocument xwpfDocument = new XWPFDocument(this.getClass().getResourceAsStream(DOCX_TEMPLATE_FILE));
-            switch (getOutputVariant()) {
-                case DISCLOSURE:
-                    fillDocument(xwpfDocument, projectLicenseInfoResults, projectName, projectVersion, licenseInfoHeaderText, false);
-                    break;
-                case REPORT:
-                    fillDocument(xwpfDocument, projectLicenseInfoResults, projectName, projectVersion, licenseInfoHeaderText, true);
-                    break;
-                default:
-                    throw new IllegalArgumentException("Unknown generator variant type: " + getOutputVariant());
+        Optional<byte[]> docxTemplateFile = CommonUtils.loadResource(DocxGenerator.class, DOCX_TEMPLATE_FILE);
+        if (docxTemplateFile.isPresent()) {
+            try {
+                XWPFDocument xwpfDocument = new XWPFDocument(new ByteArrayInputStream(docxTemplateFile.get()));
+                switch (getOutputVariant()) {
+                    case DISCLOSURE:
+                        fillDocument(xwpfDocument, projectLicenseInfoResults, projectName, projectVersion, licenseInfoHeaderText, false);
+                        break;
+                    case REPORT:
+                        fillDocument(xwpfDocument, projectLicenseInfoResults, projectName, projectVersion, licenseInfoHeaderText, true);
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Unknown generator variant type: " + getOutputVariant());
+                }
+                xwpfDocument.write(docxOutputStream);
+                docxOutputStream.close();
+            } catch (XmlException e) {
+                throw new SW360Exception("Got XmlException while generating docx document: " + e.getMessage());
+            } catch (IOException e) {
+                throw new SW360Exception("Got IOException when generating docx document: " + e.getMessage());
+            } catch (TException e) {
+                throw new SW360Exception("Error reading sw360 licenses: " + e.getMessage());
             }
-            xwpfDocument.write(docxOutputStream);
-            docxOutputStream.close();
-        } catch (XmlException e) {
-            throw new SW360Exception("Got XmlException while generating docx document: " + e.getMessage());
-        } catch (IOException e) {
-            throw new SW360Exception("Got IOException when generating docx document: " + e.getMessage());
-        } catch (TException e) {
-            throw new SW360Exception("Error reading sw360 licenses: " + e.getMessage());
+            return docxOutputStream.toByteArray();
+        } else {
+            throw new SW360Exception("Could not generate xwpf document for template: " + DOCX_TEMPLATE_FILE);
```

Comment:
this is a minor point, but I think maybe a more telling error message would be `"Could not load the template for xwpf document: " + DOCX_TEMPLATE_FILE`

Reply:
corrected.